### PR TITLE
Antivirus

### PR DESCRIPTION
**Pull request by alistairknight**

### CODE REVIEW

File: grc/templates/upload/_upload-files-section.html [hunk 1]
```diff
@@ -58,7 +58,7 @@ <h2 class="govuk-heading-m">
                 <img id="loading" src="{{ url_for('static', filename='assets/images/loading.gif') }}" width="41"
                      height="41" style="display: none; margin-bottom: 20px;" alt="">
                 <input id="documents" type="file" name="documents" multiple
-                       accept=".jpg,.jpeg,.bmp,.png,.tif,.tiff,.pdf"
+                       accept=".jpg,.jpeg,.bmp,.png,.tif,.tiff,.pdf,.txt"
```
Comment (jamesgriff) [inline grc/templates/upload/_upload-files-section.html:61]: Do we want to accept `.txt` files, or was this just for testing?

File: grc/upload/forms.py [hunk 2]
```diff
@@ -1,14 +1,15 @@
...
-from grc.utils.form_custom_validators import MultiFileAllowed, FileSizeLimit
+from grc.utils.form_custom_validators import MultiFileAllowed, fileSizeLimit, fileVirusScan
 
 
 class UploadForm(FlaskForm):
     documents = MultipleFileField(
         validators=[
-            MultiFileAllowed(['jpg', 'png', 'jpeg', 'tif', 'bmp', 'pdf'], message='Select a JPG, BMP, PNG, TIF or PDF file smaller than 10MB'),
-            FileSizeLimit(10)
+            MultiFileAllowed(['jpg', 'png', 'jpeg', 'tif', 'bmp', 'pdf', 'txt'], message='Select a JPG, BMP, PNG, TIF or PDF file smaller than 10MB'),
```
Comment (jamesgriff) [inline grc/upload/forms.py:10]: Do we want to accept `.txt` files, or was this just for testing?

File: grc/utils/form_custom_validators.py [hunk 3]
```diff
@@ -203,3 +203,40 @@ def file_length_check(form, field):
...
+
+
+def fileVirusScan(form, field):
+    if current_app.config['AV_API'] is None:
+        return
+    if (field.name not in request.files or request.files[field.name].filename == ''):
+        return
+
+    print('Scanning %s' % current_app.config['AV_API'], flush=True)
+    from pyclamd import ClamdNetworkSocket
```
Comment (jamesgriff) [inline grc/utils/form_custom_validators.py:215]: Should this import go at the top of the file?

File: grc/utils/form_custom_validators.py [hunk 4]
```diff
@@ -203,3 +203,40 @@ def file_length_check(form, field):
...
+        url = url[: url.index(':')]
+
+    cd = ClamdNetworkSocket(host=url, port=3310, timeout=None)
+    if not cd.ping():
+        print('Unable to communicate with virus scanner', flush=True)
+        return
+
+    results = cd.scan_stream(uploaded.stream.read())
+    print('results', flush=True)
+    print(results, flush=True)
```
Comment (jamesgriff) [inline grc/utils/form_custom_validators.py:233]: There's 5 `print` statements in this method.
Are they supposed to stay here for logging purposes, or were they just for testing?

Comment (alistairknight) [reply grc/utils/form_custom_validators.py:233]: Removed these

File: grc/utils/form_custom_validators.py (hunk 3 above)
Comment (alistairknight) [reply grc/utils/form_custom_validators.py:215]: As this validator is rarely used it's better in the function

**Review by jamesgriff — APPROVED** [review]

**Pull request merged**